orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy

```diff
@@ -80,6 +82,25 @@ class ApplySourceServerGroupCapacityTask extends AbstractServerGroupTask {
     }
   }
 
+  @Override
+  Moniker convertMoniker(Stage stage) {
+    try {
+      TargetServerGroupContext context = getTargetServerGroupContext(stage)
+
+      def targetServerGroup = oortHelper.getTargetServerGroup(
+        context.credentials as String,
+        context.serverGroupName as String,
+        context.region as String,
+        context.cloudProvider as String
+      ).get()
+
+      return targetServerGroup.getMoniker()
+    } catch (Exception e) {
+      log.error("Unable to apply source server group capacity (executionId: ${stage.execution.id})", e)
```

Comment:
Please add the offending server group name / region / account to this error message.

Rather than returning null on an exception, could a reasonable fallback be to construct a moniker via frigga.

The duplicate call to `clouddriver` is definitely sub-optimal given that (a) it fetches far more data than just a moniker and (b) it literally happened in the immediately preceding call (and would have also happened previously since we do have the serverGroupName in the context.

In the short-term this is probably alright, but I would like an alternative to retrieving full server groups just to get the moniker. This may end up being a _simple_ `/moniker` endpoint in clouddriver taking a few query parameters?

Reply:
I broke up the exception handling into more specific cases. I think you are right, in the case that the moniker is in the wrong place in the payload, malformed, etc. then we should fall back on frigga. In the case that there is a clouddriver problem or something else unexpected, I think falling back on frigga might obfuscate the true error downstream (this is more of an inssue for non-frigga providers.)

I agree that the double call to `clouddriver` is undesirable. I'd also love to avoid it. There are a few options here:
1. Store the response as a class member.
2. Rework `execute()` in the base class.
  a. Combine `convert()` and `converMoniker()` in 
  b. Return different types from `convert()` and use it in `convertMoniker()`
3. Put a lighter weight call into clouddriver (As you pointed out.)

I figured duplicating what was in `convert()` introduced the least amount of risk. Albeit, at the cost of a double call.

As an aside, there is a certain amount of urgency to this PR since k8s rollbacks currently do not work.

Reply:
There are no retries around the call to clouddriver, and the impact of ever having a null `Moniker` is likely to be a NPE.

In this particular task, nothing actually appears to use the moniker (`validateClusterStatus()` isn't overridden). 

I'm calling these sorts of issues out because we've run into a number of issues lately around network failures between orca <-> clouddriver (or other services) without retries.

Similarly, at scale, we've noticed unnecessary overhead from inefficient calls to fetch data from clouddriver. A lot of these calls have been optimized for `aws` but we still try and avoid them where not strictly necessary.